packages/tools-iottwinmaker/src/lib/workspace.ts

```diff
@@ -0,0 +1,182 @@
+import { getDefaultAwsClients as aws } from './aws-clients';
+
+import { WORKSPACE_ROLE_ASSUME_POLICY } from './policy/workspace-role-assume-policy';
+import { WORKSPACE_DASHBOARD_ROLE_ASSUME_POLICY } from './policy/workspace-dashboard-role-assume-policy';
+import { workspaceRolePolicyTemplate } from './policy/workspace-role-policy';
+import { workspaceDashboardRolePolicyTemplate } from './policy/workspace-dashboard-role-policy';
+import { CreatePolicyCommandOutput, CreateRoleCommandOutput } from '@aws-sdk/client-iam';
+import { CreateBucketCommandOutput } from '@aws-sdk/client-s3';
+
+/**
+ * Helper function during workspace creation to create and attach roles and policies
+ * @param uniqueId identifier for the role and policy used as suffix in name
+ * @param roleType Role or DashboardRole
+ * @param roleAssumePolicy stringified JSON of the policy
+ * @param rolePolicy stringified JSON of the role
+ * @returns promise of the roleArn
+ */
+async function createRoleAndPolicy(
+  uniqueId: string,
+  roleType: 'WorkspaceRole' | 'WorkspaceDashboardRole',
+  roleAssumePolicy: string,
+  rolePolicy: string
+) {
+  const roleName = `IoTTwinMaker${roleType}-${uniqueId}`;
+  console.log(`Creating Role ${roleName}...`);
+  const createRoleResponse: CreateRoleCommandOutput = await aws().iam.createRole({
+    RoleName: roleName,
+    Path: '/',
+    AssumeRolePolicyDocument: roleAssumePolicy,
+    Description: `${roleType} generated by AWS IoT TwinMaker Getting Started Script`,
+  });
+
+  if (!createRoleResponse.Role || !createRoleResponse.Role.Arn) {
+    throw new Error(`Failed to create role ${roleName}.`);
+  }
+
+  const roleArn: string = createRoleResponse.Role.Arn;
+  console.log(`Created Role: ${roleArn}.`);
+  const createPolicyResponse: CreatePolicyCommandOutput = await aws().iam.createPolicy({
+    PolicyName: `IoTTwinMaker${roleType}Policy-${uniqueId}`,
+    PolicyDocument: rolePolicy,
+    Description: `${roleType} Policy generated by AWS IoT TwinMaker Getting Started Script`,
+  });
+
+  const policyArn: string | undefined = createPolicyResponse.Policy?.Arn;
+  console.log(`Created Policy: ${policyArn}`);
+  await aws().iam.attachRolePolicy({
+    RoleName: roleName,
+    PolicyArn: policyArn,
+  });
+  console.log(`Attached Policy: ${policyArn} to Role: ${roleArn}.`);
+  return roleArn;
+}
+
+/**
+ * Helper function during workspace creation to create S3 bucket
+ * @param uniqueId identifier for bucket used as suffix in the name
+ * @returns promise of the role S3 bucket name and arn
+ */
+async function createWorkspaceS3Bucket(uniqueId: string) {
+  const s3BucketName = `iot-twinmaker-workspace-bucket-${uniqueId}`;
+  console.log('Creating S3 Bucket for the TwinMaker Workspace...');
+  const createBucketResponse: CreateBucketCommandOutput = await aws().s3.createBucket({ Bucket: s3BucketName });
+  console.log(`Workspace S3 Bucket created: ${createBucketResponse.Location}`);
+  console.log('Configuring CORS policy...');
```

Comment:
UX question: does this need a corresponding message to indicate it completed (similar to the "creating s3 bucket ...")

Reply:
I'm not sure I understand this one, does the current "Workspace S3 Bucket created" not indicate it completed?

Reply:
sorry, I meant I see a message like `Configuring CORS policy...` but didn't see one indicating to the customer this activity completed